### PR TITLE
Create JDK-EA builds and fix Compiler errors due to new default method in JDK-15

### PR DESCRIPTION
**Pull request by nikhilnanivadekar**
There will be failures in EBR Maven plugin. But that needs to be fixed in EBR Maven plugin. That said, the main library will be tested correctly.

### CODE REVIEW

File: .github/workflows/earlyaccess-unit.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/setup-java@v1
+        with:
+          java-version: ${{ steps.download-jdk.outputs.version }}
+          jdkFile: "${{ steps.download-jdk.outputs.file }}"
+      - name: Set Maven Wrapper
+        run: mvn -N io.takari:maven:0.7.7:wrapper -Dmaven=3.6.3
+      - name: 'JDK Version'
+        run: java --version
+      - name: Unit-Test
+        run: ./mvnw verify --no-transfer-progress
```
Comment (mohrezaei) [inline .github/workflows/earlyaccess-unit.yml:37]: create a new profile in the top level pom.xml that excludes the p2 stuff (see this [example](https://stackoverflow.com/questions/4448599/maven-modules-optional-in-pom-xml)). Call it `ea-profile`. Then specify it on this line with `-P ea-profile`.

Comment (nikhilnanivadekar) [reply .github/workflows/earlyaccess-unit.yml:37]: We need to test the p2 stuff as well. Also the profile part is not working. If you look at earlyaccess-javadoc.yml in this PR, I specifically give the modules to build and still p2 repo is getting built. That is the whole deal with EBR plugin and I do not know how to solve that. I am working on fixing the problem with EBR plugin itself. PR raised there: https://github.com/eclipse/ebr/pull/40

**Comment by nikhilnanivadekar** [comment]
@mohrezaei @donraab is this fine to merge?
The p2 failures are going to be painful, but we can solve that as step 2. I would rather merge this and cherry-pick the compiler error fixes and release 10.4.0.
JDK-15- Final RC will be released on 20th August, so would like to close out the release before that.
Thoughts?

**Comment by nikhilnanivadekar** [comment]
After upgrading the EBR plugin to 1.3.0-SNAPSHOT the builds are working fine. My fix was merged here: https://github.com/eclipse/ebr/pull/40

Can this PR be merged?

**Pull request merged**